aspnetcore/blazor/file-uploads.md

```diff
@@ -92,6 +85,38 @@ await blobContainerClient.UploadBlobAsync(
 
 A component that receives an image file can call the <xref:Microsoft.AspNetCore.Components.Forms.BrowserFileExtensions.RequestImageFileAsync%2A?displayProperty=nameWithType> convenience method on the file to resize the image data within the browser's JavaScript runtime before the image is streamed into the app. Use cases for calling <xref:Microsoft.AspNetCore.Components.Forms.BrowserFileExtensions.RequestImageFileAsync%2A> are most appropriate for Blazor WebAssembly apps.
 
+## File size limit
+
+:::moniker range=">= aspnetcore-6.0"
+
+:::zone pivot="server"
+
+There's no a limit on the maximum file size for the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component in Blazor Server apps.
+
+:::zone-end
+
+:::zone pivot="webassembly"
+
+There's no a limit on the maximum file size for the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component in Blazor WebAssembly apps. However, `BrowserHttpHandler.SendAsync` ([reference source](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs)) reads the file's bytes into a single JavaScript array buffer when marshalling the data from JavaScript to C#, so large file uploads (> 250 MB) may fail. For more information, see [Net6P7: Blazor WASM can't upload large files (500MB, 1GB, 2GB) (dotnet/aspnetcore #35899)](https://github.com/dotnet/aspnetcore/issues/35899).
```

Comment:
I would start the sentence off with, "Technically, there is no limit..." I'm afraid that someone will read the first sentence and then stop. Or not understand the contradicting second part. Actually, thinking about it more, what are you even trying to accomplish with the first sentence? It sounds like you're trying to make a sales pitch and attempt to mislead the customer/developer.

Therefore, I would instead completely change the first sentence to something like:
> In Blazor WebAssembly, when using the \<xref:Microsoft.AspNetCore.Components.Forms.InputFile> component, the maximum file size that can be uploaded will be limited by the amount of available process memory of the web browser. This is due to how `BrowserHttpHandler.SendAsync` ([reference source](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs)) reads the file's bytes into a single JavaScript array buffer when marshalling the data from JavaScript to C#, so large file uploads (> 250 MB) may fail. For more information, see [Net6P7: Blazor WASM can't upload large files (500MB, 1GB, 2GB) (dotnet/aspnetcore #35899)](https://github.com/dotnet/aspnetcore/issues/35899).

Note that if you don't take either of my suggesting here, you should at least change the beginning of the sentence again from "There's no **a** limit..." to "There's no limit...".

Reply:
I don't care for double prepositional phrase opening constructions because in many cases the 2nd modifies the object of the first, scoping it in some essential way, not merely introducing parenthetical info. However, I'm fine with a fix here. I'll try to shape it along the lines of your suggestion.

Reply:
Also ... that's a bit inaccurate because it conflates the prior limit that was hardcoded into the `InputFile` component, which they fixed later.

I'll work on it and see what we can do here. *Stand-by* ......

Reply:
> what are you even trying to accomplish with the first sentence?

They fixed it at 6.0. Anyone who knows that it was limited in the past, perhaps from a PU issue that they might have come across while researching this, should have a clear statement that the limit isn't related to the component.